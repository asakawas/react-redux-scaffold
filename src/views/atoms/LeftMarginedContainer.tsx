import React from 'react';
import Container, { ContainerProps } from '@material-ui/core/Container';
import styled from 'styled-components';
import { spacing, SpacingProps } from '@material-ui/system';

/**
 * The type of props of LeftMarginedContainer.
 */
export type LeftMarginedContainerProps = Readonly<
  ContainerProps &
    SpacingProps & {
      /** The left margin (in px) of the container. */
      marginLeft: number;
    }
>;

const LeftMarginedContainer = styled(({ marginLeft, ...props }: LeftMarginedContainerProps) => (
  // Remove LeftMarginedContainerProps to suppress a runtime warning from react-dom.
  // eslint-disable-next-line react/jsx-props-no-spreading
  <Container {...props} />
))`
  ${spacing}
  margin-left: ${({ marginLeft }: LeftMarginedContainerProps) => marginLeft}px;
  width: calc(100% - ${({ marginLeft }: LeftMarginedContainerProps) => marginLeft}px);
`;

export default LeftMarginedContainer;
