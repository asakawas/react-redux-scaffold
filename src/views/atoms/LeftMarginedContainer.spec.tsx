import React from 'react';
import { render } from '@testing-library/react';
import LeftMarginedContainer from './LeftMarginedContainer';

describe('LeftMarginedContainer', () => {
  test('is rendered as expected', () => {
    const { asFragment } = render(
      <LeftMarginedContainer marginLeft={200}>
        <div />
      </LeftMarginedContainer>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('has specified margin in its style attribute', () => {
    const renderResult = render(
      <LeftMarginedContainer marginLeft={123} data-testid="test">
        <div />
      </LeftMarginedContainer>,
    );

    const tree = renderResult.getByTestId('test');
    expect(tree).toHaveStyleRule('margin-left', '123px');
    expect(tree).toHaveStyleRule('width', 'calc(100% - 123px)');
  });
});
