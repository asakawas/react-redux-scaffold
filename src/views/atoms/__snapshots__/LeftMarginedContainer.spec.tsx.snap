// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftMarginedContainer is rendered as expected 1`] = `
<DocumentFragment>
  .c0 {
  margin-left: 1600px;
  margin-left: 200px;
  width: calc(100% - 200px);
  text-align: center;
}

<div
    class="MuiContainer-root c0 MuiContainer-maxWidthLg"
  >
    <div />
  </div>
</DocumentFragment>
`;
